spec/std/enumerable_spec.cr

```diff
@@ -92,6 +92,58 @@ describe "Enumerable" do
     end
   end
 
+  describe "chunk" do
+    it "works" do
+      [1].chunk { true }.to_a.should eq [{true, [1]}]
+      [1, 2].chunk { false }.to_a.should eq [{false, [1, 2]}]
+      [1, 1, 2, 3, 3].chunk(&.itself).to_a.should eq [{1, [1, 1]}, {2, [2]}, {3, [3, 3]}]
+      [1, 1, 2, 3, 3].chunk(&.class).to_a.should eq [{Int32, [1, 1, 2, 3, 3]}]
+      [1, 1, 2, 3, 3].chunk(&.<=(2)).to_a.should eq [{true, [1, 1, 2]}, {false, [3, 3]}]
+      (0..10).chunk(&./(3)).to_a.should eq [{0, [0, 1, 2]}, {1, [3, 4, 5]}, {2, [6, 7, 8]}, {3, [9, 10]}]
+    end
+
+    it "works with block" do
+      res = [] of Tuple(Bool, Array(Int32))
+      [1, 2, 3].chunk { |x| x < 3 }.each { |k, v| res << {k, v} }
```

Comment:
Can you use `|(k, v)|` for now? Auto-unpacking is only available in the next version, with this the current specs can't be compiled anymore with 0.17.4.